land_grab_2/stl_dataset/step_2_5/get_cessions.py

```diff
@@ -0,0 +1,278 @@
+import logging
+import os
+from pathlib import Path
+from typing import List
+import warnings
+
+import pandas as pd
+import geopandas as gpd
+import numpy as np
+
+from land_grab_2.stl_dataset.step_1.constants import WGS_84
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger(__name__)
+
+
+def area_join(
```

Comment:
This function is essentially an area-weighted one-to-one spatial join. It only joins the record from `source_df` that has the largest intersection by area with the record from `target_df`. This emulates the behavior of QGIS'  **Join attributes by location** → _Take attributes of the feature with the largest overlap only - one to one_ option.